opm/simulators/timestepping/AdaptiveTimeStepping_impl.hpp

```diff
@@ -351,12 +360,7 @@ namespace Opm {
                 const double newTimeStep = restart_factor_ * dt;
                 // we need to revise this
                 substepTimer.provideTimeStepEstimate( newTimeStep );
-                if( solver_verbose_ ) {
```

Comment:
I think this should be kept, and added to. The missing thing is an explanation for why we are cutting the timestep. My suggestion is:
1. Create a variable (`std::string cause_of_failure = "";`) before the try-block (where substepReport is defined).
2. In the catch blocks, do `cause_of_failure = e.what();`.
3. Down here, add  `"Cause for convergence failure was: " + cause_of_failure` to the `msg` string.

If we want more refined cause strings we can add two new catch-blocks, one for LinearSolverProblem and another for TooManyIterations, and write anything we want there (instead of using the what()). If we want to extract the number of iterations that failed we should add that number as a variable to the TooManyIterations exception class rather than extracting it from the string.